code/modules/mob/living/carbon/human/human_defense.dm

```diff
@@ -325,24 +311,19 @@ emp_act
 		hitpush = 0
 		skipcatch = 1
 		blocked = 1
-	/*else if(I)
+	else if(I)
 		if(I.throw_speed >= EMBED_THROWSPEED_THRESHOLD)
-			if(!I.is_robot_module())
-				var/armor = run_armor_check(affecting, "melee", "Your armor has protected your [hit_area].", "Your armor has softened hit to your [hit_area].", I.armour_penetration) //I guess "melee" is the best fit here
-				var/sharp = is_sharp(I)
-				var/damage = throwpower * (I.throw_speed / 5)
-				if(armor)
-					damage /= armor + 1
-
-				//blunt objects should really not be embedding in things unless a huge amount of force is involved
-				var/embed_chance = sharp? damage / I.w_class : damage/(I.w_class * 3)
-				var/embed_threshold = sharp? 5 * I.w_class : 15 * I.w_class
-
-				//Sharp objects will always embed if they do enough damage.
-				//Thrown sharp objects have some momentum already and have a small chance to embed even if the damage is below the threshold
-
-				if(((sharp && prob(damage / (10 * I.w_class) * 100)) || (damage > embed_threshold && prob(embed_chance))) && (I.no_embed == 0))
-					affecting.embed(I)*/
+			if(can_embed(I))
+				if(prob(I.embed_chance))
+					throw_alert("embeddedobject", /obj/screen/alert/embeddedobject)
+					var/obj/item/organ/external/L = pick(bodyparts)
+					L.embedded_objects |= I
+//					I.add_mob_blood(src)//it embedded itself in you, of course it's bloody!

```

Comment:
Adding in commented out code, I know; this is because of the blood refactor; will uncomment once its merged.